src/librustc_lint/builtin.rs

```diff
@@ -272,6 +272,22 @@ impl EarlyLintPass for UnsafeCode {
                 })
             }
 
+            ast::ItemKind::Fn(..) => {
+                if attr::contains_name(&it.attrs, sym::no_mangle) {
+                    self.report_unsafe(cx, it.span, |lint| {
+                        lint.build("declaration of a `no_mangle` function").emit();
+                    })
+                }
+            }
+
+            ast::ItemKind::Static(..) => {
+                if attr::contains_name(&it.attrs, sym::no_mangle) {
+                    self.report_unsafe(cx, it.span, |lint| {
+                        lint.build("declaration of a `no_mangle` static").emit();
+                    })
+                }
+            }
```

Comment:
I believe these are the only item kinds on which `#[no_mangle]` makes a difference, it seems like you can stick in a lot of places where it's ignored. Should this check everywhere, or only locations where the attribute actually affects compilation?

Reply:
Only in places that actually affect the resulting binary. The others should receive warnings or errors about the non effect already. It is good practice to avoid emitting multiple errors for the same cause. If that already happens, we try to silence the less informative one(s).